rope_.d

```diff
@@ -74,19 +74,19 @@ template Rope(T,S=void)if(is(S==void) || is(typeof((S a,S b)=>a.combine(b)))){
 		static if(segtree){
 			@property value(){ return isArray()?tree?tree.value:S.init:rope.value; }
 			private this(T[] array, SliceSegTree* tree){
-				if(array==[]) return;
 				this.array = array;
+				if(array==[]) return;
 				this.tree = tree;
 			}
 		}
 
 		private this(RopeImpl* rope){ this.rope = rope; }
 		invariant(){ assert(cast(void*)rope is array.ptr); }
-		private union{
+		/*private*/ union{
```

Comment:
This change is the reason for the segmentation fault. I didn't notice this for a while because it looks so benign.
Is there a better solution? (A workaround for the problem is to rename the 'array' member.)